File: design-patterns/go/decorator/ui_element/background_decorator.go
// background_decorator.go

package main

import "fmt"

type BackgroundDecorator struct {
	*UIDecorator
}

func NewBackgroundDecorator(uiElement UIElement) (backgroundDecorator *BackgroundDecorator) {
	backgroundDecorator = &BackgroundDecorator{}
	backgroundDecorator.UIDecorator = NewUIDecorator(uiElement)
	return
}

func (backgroundDecorator *BackgroundDecorator) Draw() {
	// Can perform any additional task anywhere in the method

	backgroundDecorator.UIDecorator.Draw()

	// Write code to add background to the element
	fmt.Println("Adding Background to the element")
}

File: design-patterns/php/decorator/uielement/BackgroundDecorator.php
<?php
// BackgroundDecorator.php

namespace BigBoxCode\DesignPattern\Decorator\UiElement;

class BackgroundDecorator extends UIDecorator {
    public function __construct(UIElement $uiElement) {
        parent::__construct($uiElement);
    }

    public function draw(): void {
        parent::draw();
        
        echo "Adding Background to the element\n";
    }
}